dali/base/dasds.cpp

```diff
@@ -1407,17 +1427,36 @@ class CDeltaWriter : implements IThreaded
         CriticalBlock b(pendingCrit);
         addToQueue(item);
     }
-    void clear()
+    void flush()
```

Comment:
This should be commented because it isn't clear what the different flags mean etc.  Something like
//ensure that all pending transactions have been flushed to the file.

if (waiting)
{
   // The writer thread is actively waiting - if there is data to write ensure it is written
}
else
{
   //delta is currently being written.  Wait until all the data is written.

Reply:
will update with some comments.